mlx/backend/cuda/conv.cpp

```diff
@@ -263,78 +361,104 @@ void Convolution::eval_gpu(const std::vector<array>& inputs, array& out) {
 
   encoder.set_input_array(in);
   encoder.set_input_array(wt);
-  encoder.set_output_array(out);
-
-  auto backend_type = CUDNN_BACKEND_OPERATION_CONVOLUTION_FORWARD_DESCRIPTOR;
-  auto cudnn_type = dtype_to_cudnn_type(in.dtype());
+  encoder.set_output_array(out_);
```

Comment:
Should `out` or `out_` be passed here? 🤔 I think the final output `out_` should be passed but want to be sure.

Reply:
They have the same data so it actually doesn't matter in this case. But in general the rule is set the array (or arrays) whose data you write to in the subsequent kernel.